### PR TITLE
feat: add API of replace option for copy

### PR DESCRIPTION
**Pull request by liuxh0**
Add API functionality to specify replace option when copying dataset.

If implementation is okay, I can also add for enq option in the same way.

Related to #579

### CODE REVIEW

**Comment by CForrest97** [comment]
Hi @liuxh0, thanks for adding in the replace option! There's a couple of things I would suggest adding to this before it can be PR'd though. 
Could you unit test adding in replace to confirm the `copyExpectStringSpy` receives the option `replace=true`. And add system tests to confirm that setting `replace=true`has the intended affect (i.e without `replace=true` you get an error but when trying to replace a member with `replace=true` it works)
Secondly, if you look at some of the other API methods, the options are passed as a separate parameter. Could you create an `ICopyDataSetOptions` interface which has an optional replace boolean? `IDeleteDatasetOptions` and `Delete.dataSet` should be good examples on how to handle options

**Comment by zFernand0** [comment]
@liuxh0,
I really appreciate you using `Conventional Commit` standards. :+1:

@MarkAckert @jackjia-ibm @MikeBauerCA,
Any ideas how we can make this a standard across Zowe?

**Comment by liuxh0** [comment]
Hi @CForrest97, thanks for your suggestions. I am new to the project.

Now I have moved the option to `ICopyDataSetOptions` and added some unit tests. 

However, `custom_properties.yaml` is missing because I don't have access to any test system. Can you help me to get access to test system? After that I will add system tests.

**Comment by liuxh0** [comment]
@CForrest97 @zFernand0 
I tried to also add some system tests. As I said above, I can't test locally. 
And then I also realized that Jenkins doesn't run system tests. I'm not sure if these tests pass or not. 
Can you help me moving this forward? Thanks!

**Comment by CForrest97** [comment]
@liuxh0 unfortunately getting a test system isn't too straight forward. It may be worth checking out the #zowe-onboarding channel on the open mainframe project's slack workspace. There's people like Joe Winchester who could help out.
In the mean time, I can fork your repo and confirm the system tests work

**Comment by liuxh0** [comment]
@CForrest97 
That would be great. Thanks!

File: packages/zosfiles/src/api/methods/copy/doc/ICopyDatasetOptions.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+/**
+ * This interface defines the options that can be sent into the copy data set function.
+ */
+export interface ICopyDatasetOptions {
```
Comment (zFernand0) [inline packages/zosfiles/src/api/methods/copy/doc/ICopyDatasetOptions.ts]: I believe this Interface could be extended to support all options on the payload.
However there are more changes to the payload on #599.
I'd suggest merging this in, and making all necessary updates on the other PR :)

File: packages/zosfiles/__tests__/__system__/api/methods/copy/Copy.system.test.ts [hunk 2]
```diff
@@ -216,5 +216,73 @@ describe("Copy", () => {
...
+
+                try {
+                    response = await Copy.dataSet(
+                        REAL_SESSION,
+                        { dataSetName: fromDataSetName, memberName: file1 },
+                        { dataSetName: toDataSetName, memberName: file2 },
+                        { replace: true }
+                    );
+                    contents1 = await Get.dataSet(REAL_SESSION, `${fromDataSetName}(${file1})`);
+                    contents2 = await Get.dataSet(REAL_SESSION, `${toDataSetName}(${file2})`);
```
Comment (zFernand0) [inline packages/zosfiles/__tests__/__system__/api/methods/copy/Copy.system.test.ts]: :+1: Thanks for checking the contents of the replaced dataset 👍

File: packages/zosfiles/__tests__/api/methods/copy/Copy.unit.test.ts [hunk 3]
```diff
@@ -189,6 +189,46 @@ describe("Copy", () => {
...
+                });
+                it("should contain replace with value false in payload", async () => {
+                    await Copy.dataSet(
+                        dummySession,
+                        { dataSetName: fromDataSetName },
+                        { dataSetName: toDataSetName },
+                        { replace: false }
+                    );
+
+                    expect(copyExpectStringSpy).toHaveBeenCalledTimes(1);
```
Comment (zFernand0) [inline packages/zosfiles/__tests__/api/methods/copy/Copy.unit.test.ts]: :+1: cool use of the top level before each

**Comment by zFernand0** [comment]
Please port to `lts-incremental`

**Comment by zFernand0** [comment]
Bypassing vulnerability check.
Please reference https://github.com/zowe/imperative/issues/327

**Comment by liuxh0** [comment]
@zFernand0 Thanks for letting this PR through. 
I have seen there is a PR merging this to `lts-incremental`. (Sorry I saw your comment too late.)
I think now I'll continue with other options mentioned in #579.

**Comment by zFernand0** [comment]
No problem at all, thanks for implementing : )

**Pull request merged**